### PR TITLE
Fix for throughput impact on non-consumption plans

### PR DESCRIPTION
**Pull request by safihamid**

### CODE REVIEW

File: src/WebJobs.Script.WebHost/DependencyInjection/DryIoc/Container.cs [hunk 1]
```diff
@@ -6574,7 +6580,8 @@ protected virtual Expr ApplyReuse(Expr serviceExpr, Request request)
                 !request.TracksTransientDisposable &&
                 !request.IsWrappedInFunc())
             {
-                var factoryDelegate = serviceExpr.CompileToFactoryDelegate();
+                var container = request.Container as Container;
```
Comment (alrod) [inline src/WebJobs.Script.WebHost/DependencyInjection/DryIoc/Container.cs]: Do we need check for "null" here?

**Comment by alrod** [comment]
please squash to one commit

File: src/WebJobs.Script.WebHost/DependencyInjection/DryIoc/Container.cs (hunk 1 above)
Comment (safihamid) [reply src/WebJobs.Script.WebHost/DependencyInjection/DryIoc/Container.cs]: it will never be null but will add the check just to be sure 100%

Comment (safihamid) [reply src/WebJobs.Script.WebHost/DependencyInjection/DryIoc/Container.cs]: actually it will never been null.

**Pull request merged**